src/minio/models/minio_config.py

```diff
@@ -0,0 +1,64 @@
+import os
+from typing import Annotated
+
+from pydantic import BaseModel, ConfigDict, Field
+
+CDM_DEFAULT_BUCKET = os.getenv("CDM_DEFAULT_BUCKET", "cdm-lake")
+CDM_DEFAULT_WAREHOUSE_PREFIX = os.getenv("CDM_DEFAULT_WAREHOUSE_PREFIX", "warehouse")
+
+
+class MinIOConfig(BaseModel):
+    """
+    MinIO configuration model.
+
+    This model contains the essential configuration needed for MinIO operations.
+    """
+
+    model_config = ConfigDict(
+        str_strip_whitespace=True, validate_assignment=True, extra="forbid"
+    )
+
+    # Core connection settings
+    endpoint: Annotated[
+        str,
+        Field(
+            description="MinIO server endpoint URL",
+            examples=["http://localhost:9002", "https://minio.example.com"],
+        ),
+    ]
+    access_key: Annotated[
+        str,
+        Field(
+            description="MinIO access key for authentication",
+            examples=["minio", "admin"],
+        ),
+    ]
+    secret_key: Annotated[
+        str,
+        Field(
+            description="MinIO secret key for authentication",
+            examples=["minio123", "password123"],
+        ),
+    ]
```

Comment:
These would allow empty strings; I think a min length param would fix that although I don't know if that applies before or after whitespace is stripped

Reply:
:+1: